espwrap/base.py

```diff
@@ -71,6 +71,18 @@ def __init__(self, subject='', from_addr='', text='', html='',
         self.send_at = None
         self.metadata = metadata or {}
 
+    @property
+    def tags(self):
+        return self.__tags
+
+    @tags.setter
+    def tags(self, values):
+        if not values:
+            self.__tags = []
+        for tag in values:
+            if tag not in self.tags:
+                self.__tags.append(tag)
```

Comment:
Why not use a `set` to store tags? Seems like the right data structure since you want to avoid duplicates

Reply:
Mostly just to preserve order.

Reply:
Makes sense